Dockerfile

```
@@ -15,13 +15,26 @@ ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8
 
 FROM common AS sgx
+ARG PSW_VERSION=2.13.103.1
+ARG DCAP_VERSION=1.10.103.1
 RUN apt update && \
-    apt install -y --no-install-recommends libsgx-dcap-ql libsgx-dcap-ql-dev libsgx-dcap-quote-verify libsgx-enclave-common libsgx-urts az-dcap-client && \
+    apt install -y --no-install-recommends \
+    az-dcap-client \
+    libsgx-dcap-ql=$DCAP_VERSION-bionic1 \
+    libsgx-dcap-ql-dev=$DCAP_VERSION-bionic1 \
+    libsgx-pce-logic=$DCAP_VERSION-bionic1 \
+    libsgx-qe3-logic=$DCAP_VERSION-bionic1 \
+    libsgx-dcap-quote-verify=$DCAP_VERSION-bionic1 \
+    libsgx-enclave-common=$PSW_VERSION-bionic1 \
+    libsgx-urts=$PSW_VERSION-bionic1 \
     apt clean && apt autoclean
+# rename the softlink created by libsgx-dcap-default-qpl to avoid issues with az-dcap-client
+RUN mv /usr/lib/x86_64-linux-gnu/libdcap_quoteprov.so.1 /usr/lib/x86_64-linux-gnu/libdcap_quoteprov.so.1.intel
 
 FROM sgx AS sgx-dev
+ARG PSW_VERSION=2.13.103.1
```

```suggestion
ARG PSW_VERSION
```
suggest declaring the args in the stages, but defining them once before the stages as demonstrated here https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact